ids/format_test.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package ids
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestFormat(t *testing.T) {
+	type (
+		idInterface interface {
+			String() string
+			Hex() string
+		}
+		test struct {
+			id   idInterface
+			want map[string]string // format -> output
+		}
+	)
+	makeTestCase := func(id idInterface) test {
+		return test{
+			id: id,
+			want: map[string]string{
+				"%v":  id.String(),
+				"%s":  id.String(),
+				"%q":  `"` + id.String() + `"`,
+				"%x":  id.Hex(),
+				"%#x": `0x` + id.Hex(),
+			},
+		}
+	}
+
+	tests := []test{
+		makeTestCase(ID{}),
+		makeTestCase(GenerateTestID()),
+		makeTestCase(GenerateTestID()),
+		makeTestCase(ShortID{}),
+		makeTestCase(GenerateTestShortID()),
+		makeTestCase(GenerateTestShortID()),
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.id.String(), func(t *testing.T) {
+			for format, want := range tt.want {
+				if got := fmt.Sprintf(format, tt.id); got != want {
+					//nolint:forbidigo // require.Equal() is inappropriate as it unnecessarily stops future tests and `assert.Equal()` isn't allowed
+					t.Errorf("fmt.Sprintf(%q, %T) got %q; want %q", format, tt.id, got, want)
+				}
+			}
+		})
+	}
+}
+
+func BenchmarkFormat(*testing.B) {
+	// %q uses a []byte so this is just to demonstrate that it's on the stack
+	// otherwise someone, not naming any names, might want to "fix" it.
+	_ = fmt.Sprintf("%q", ID{})
+	_ = fmt.Sprintf("%q", ShortID{})
+}
```

Comment:
What is this doing?

Reply:
It's purely so you can run it and see that there are no allocs. Very happy to remove it entirely.

Reply:
```suggestion
func BenchmarkFormat(b *testing.B) {
	// %q uses a []byte so this is just to demonstrate that it's on the stack
	// otherwise someone, not naming any names, might want to "fix" it.
	b.Run("ID", func(b *testing.B) {
		for range b.N {
			_ = fmt.Sprintf("%q", ID{})
		}
	})
	b.Run("ShortID", func(b *testing.B) {
		for range b.N {
			_ = fmt.Sprintf("%q", ShortID{})
		}
	})
}
```

Outputs:
```
goos: darwin
goarch: arm64
pkg: github.com/ava-labs/avalanchego/ids
cpu: Apple M2 Max
BenchmarkFormat/ID-12 			 3499521	       350.7 ns/op	     272 B/op	       6 allocs/op
BenchmarkFormat/ShortID-12         	 3796848	       309.6 ns/op	     176 B/op	       6 allocs/op
PASS
coverage: 10.9% of statements in all
```